subprojects/core/src/main/java/org/gradle/internal/build/BuildOperationFiringBuildWorkPreparer.java

```diff
@@ -96,8 +98,13 @@ public void run(BuildOperationContext buildOperationContext) {
             Set<Task> filteredTasks = contents.getFilteredTasks();
             QueryableExecutionPlan.ScheduledNodes scheduledWork = contents.getScheduledNodes();
 
-            buildOperationContext.setResult(new CalculateTaskGraphBuildOperationType.Result() {
-                private final List<CalculateTaskGraphBuildOperationType.PlannedTask> taskPlan = toPlannedTasks(scheduledWork);
+            List<PlannedNode> plannedNodes = toPlannedNodes(scheduledWork);
+            List<PlannedTask> plannedTasks = plannedNodes.stream()
```

Comment:
I wouldn't calculate the `plannedTasks` from the `plannedNodes` when we create the build operation but on-demand when the fields are queried. That way we may avoid the overhead of always creating this list of `plannedTasks`.

Reply:
That's a good idea. For newer versions of the GE plugin, it wouldn't even request the task nodes.